quotecache/quotecache.go

```diff
@@ -27,26 +27,36 @@ func (q Quote) IsExpired() bool {
 	return time.Now().After(expiry)
 }
 
+var QuoteCache = make(map[string]map[string]Quote)
+
 // Global to store cached responses.
 // Maps stock name -> Quote
 var cache = make(map[string]Quote)
 
 // GetQuote : Gets the current value of the stock, hitting the local cache if it can.
 func GetQuote(userID, stock string) (Quote, error) {
 	// check if the value is in cache
-	quote, found := cache[stock]
 
-	if !found || quote.IsExpired() {
+	var userQuote Quote
+	userMap := QuoteCache[stock]
+	userQuote, found := userMap[userID]
+	isFresh := userQuote.Timestamp.Unix() - time.Now().Unix() <= 60
```

Comment:
Sure, but why does userQuote.IsExpired() use 45s instead of the 60s? Is that a mistake?